### ISSUE TITLE
get/set/pause/resume current playback state for audio componentsa

### ISSUE THREAD

**Issue by subsoap** [defold/defold#7601]
For rhythm games need to be able to sync gameplay state with audio state. Since audio is playing on its own thread (which is generally better than it was before in terms of playback quality) it can desync from the timing of the game. This is most easily seen in HTML5 games where you can switch between the game's tab and another tab a few times and already the desynced state is very evident. 

Rhythm games often need the audio and the game to sync. When game loses focus we will want to pause the game and then when the player wants to resume we want to resume the audio at the proper timing that the game thinks it is at. In some other situations we simply want to either sync the gameplay to the audio (where it's more important to keep the audio consistent) or the audio to the gameplay.

```
sound.play() -- already implemented
sound.stop() -- already implemented
sound.get() -- returns table for a sound component, current time, total time, gain, pan, speed
sount.set() -- set current time / gain / pan / speed
sound.pause() -- pause playback at current time
sound.resume() -- optionally include time to resume at that time
```

**Comment by subsoap** [comment]
Bonus request, allow us to get data from the audio that would enable us to create visualizers / create elements which react to audio more dynamically.

**Comment by Jhonnyg** [comment]
> Bonus request, allow us to get data from the audio that would enable us to create visualizers / create elements which react to audio more dynamically.

What data do you mean? Raw samples? That feels like it should live in another issue / feature request I think :)

**Comment by subsoap** [comment]
> That feels like it should live in another issue / feature request I think :)

You're right! But maybe some part of it would be appropriate for this issue since it could be part of a larger focus to improve audio features.

> What data do you mean?

Main thing people will ask for is a way to generate visualizers that are from audio data. 

Or a way to get snapshots of data directly from the sound component that can then be analyzed / made useful with a script or extension. It's possible to just read the raw audio file with an extension currently probably but since there is an issue with sound desyncing from the game sim and that there is no way to resync even though this can be done it can't actually be used in production currently.

Here's an example from FMOD/Unity:

https://fmod.com/docs/2.02/unity/examples-spectrum-analysis.html
